File: Mystack.java
import java.util.Scanner;
import java.util.Stack;

public class StringReverser {
    
    public static String readString() {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter a word or sentence: ");
        String input = scanner.nextLine();
        return input;
    }
    
    public static Stack<Character> pushLettersToStack(String input) {
        Stack<Character> stack = new Stack<>();
        for (char c : input.toCharArray()) {
            stack.push(c);
        }
        return stack;
    }
    
    public static Stack<Character> reverseStack(Stack<Character> s) {
        Stack<Character> reversed = new Stack<>();
        while (!s.isEmpty()) {
            reversed.push(s.pop());
        }
        return reversed;
    }
    
    public static String stackToString(Stack<Character> s) {
        StringBuilder result = new StringBuilder();
        while (!s.isEmpty()) {
            result.append(s.pop());
        }
        return result.toString();
    }
    
    public static void main(String[] args) {
        String input = readString();
        Stack<Character> stInput = pushLettersToStack(input);
        String reversed = stackToString(reverseStack(stInput));
        System.out.println("Reversed: " + reversed);
    }
}

File: MyStack.cpp
#include <iostream>
#include <stack>
#include <string>

using namespace std;

string ReadString()
{
    string input;
    cout << "Enter a word or sentence: ";
    getline(cin, input);

    return input;
}

stack<char> PushLettersToStack(string& input)
{
    stack<char> s;
    for (char c : input) {
        s.push(c);
    }
    return s;
}

stack <char> ReverseStack(stack <char> s)
{
    stack <char> reversed;
    while (!s.empty()) {
        reversed.push(s.top());
        s.pop();
    }
    return reversed;
}

string StackToString(stack <char> s)
{
    string result;
    while (!s.empty()) {
        result += s.top();
        s.pop();
    }
    return result;
}

int main() {

    string input = ReadString();
    stack <char> stInput = PushLettersToStack(input);
    string reversed = StackToString(ReverseStack(stInput));
    cout << "Reversed: " << reversed << endl;

    return 0;
}